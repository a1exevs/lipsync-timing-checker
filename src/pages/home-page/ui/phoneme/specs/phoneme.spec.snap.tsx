import { render } from '@testing-library/react';
import React from 'react';

import Phoneme from 'src/pages/home-page/ui/phoneme/phoneme';

describe('Phoneme', () => {
  const baseProps = {
    id: 'p1',
    phoneme: 'a',
    leftPercent: 10,
    widthPercent: 20,
    onResizeStart: jest.fn(),
    onChainResizeStart: jest.fn(),
    onMoveStart: jest.fn(),
    movingInProgress: false,
  };

  it('matches snapshot (default)', () => {
    const { asFragment } = render(<Phoneme {...baseProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('matches snapshot (without borders)', () => {
    const { asFragment } = render(<Phoneme {...baseProps} withoutLeftBorder withoutRightBorder />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('matches snapshot (movingInProgress)', () => {
    const { asFragment } = render(<Phoneme {...baseProps} movingInProgress />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('matches snapshot (hideChainResizer)', () => {
    const { asFragment } = render(<Phoneme {...baseProps} hideChainResizer />);
    expect(asFragment()).toMatchSnapshot();
  });
});
