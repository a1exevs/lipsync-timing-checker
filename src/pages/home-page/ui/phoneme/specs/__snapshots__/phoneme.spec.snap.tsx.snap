// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Phoneme matches snapshot (default) 1`] = `
<DocumentFragment>
  <div
    class="absolute text-black text-center border border-black rounded cursor-grab whitespace-nowrap bg-[#ffff99] hover:bg-[#ffff99aa] transition-colors duration-200 ease-in-out"
    style="width: 20%; left: 10%;"
  >
    <div
      class="absolute top-0 h-full cursor-col-resize left-0"
      style="z-index: 2; width: 5px;"
    />
    <div
      class="select-none"
    >
      a
    </div>
    <div
      class="absolute top-0 h-full cursor-col-resize right-0"
      style="z-index: 2; width: 5px;"
    />
    <div
      class="absolute top-0 h-full cursor-col-resize"
      style="z-index: 2; width: 5px; right: -2.5px;"
    />
  </div>
</DocumentFragment>
`;

exports[`Phoneme matches snapshot (hideChainResizer) 1`] = `
<DocumentFragment>
  <div
    class="absolute text-black text-center border border-black rounded cursor-grab whitespace-nowrap bg-[#ffff99] hover:bg-[#ffff99aa] transition-colors duration-200 ease-in-out"
    style="width: 20%; left: 10%;"
  >
    <div
      class="absolute top-0 h-full cursor-col-resize left-0"
      style="z-index: 2; width: 5px;"
    />
    <div
      class="select-none"
    >
      a
    </div>
    <div
      class="absolute top-0 h-full cursor-col-resize right-0"
      style="z-index: 2; width: 5px;"
    />
  </div>
</DocumentFragment>
`;

exports[`Phoneme matches snapshot (movingInProgress) 1`] = `
<DocumentFragment>
  <div
    class="absolute text-black text-center border border-black rounded cursor-grab whitespace-nowrap bg-[#ffff99] hover:bg-[#ffff99aa] transition-colors duration-200 ease-in-out shadow-[0_4px_8px_0_white] u-translate-y-[5px] cursor-grabbing transition-transform transition-shadow duration-200 ease-in-out"
    style="width: 20%; left: 10%;"
  >
    <div
      class="absolute top-0 h-full cursor-col-resize left-0"
      style="z-index: 2; width: 5px;"
    />
    <div
      class="select-none"
    >
      a
    </div>
    <div
      class="absolute top-0 h-full cursor-col-resize right-0"
      style="z-index: 2; width: 5px;"
    />
    <div
      class="absolute top-0 h-full cursor-col-resize"
      style="z-index: 2; width: 5px; right: -2.5px;"
    />
  </div>
</DocumentFragment>
`;

exports[`Phoneme matches snapshot (without borders) 1`] = `
<DocumentFragment>
  <div
    class="absolute text-black text-center border border-black rounded cursor-grab whitespace-nowrap bg-[#ffff99] hover:bg-[#ffff99aa] transition-colors duration-200 ease-in-out border-l-0 border-r-0"
    style="width: 20%; left: 10%;"
  >
    <div
      class="absolute top-0 h-full cursor-col-resize left-0"
      style="z-index: 2; width: 5px;"
    />
    <div
      class="select-none"
    >
      a
    </div>
    <div
      class="absolute top-0 h-full cursor-col-resize right-0"
      style="z-index: 2; width: 5px;"
    />
    <div
      class="absolute top-0 h-full cursor-col-resize"
      style="z-index: 2; width: 5px; right: -2.5px;"
    />
  </div>
</DocumentFragment>
`;
